esp/platform/espcontext.hpp

```diff
@@ -31,6 +31,61 @@
 #include "esp.hpp"
 #include "esphttp.hpp"
 
+#define SESSION_SDS_LOCK_TIMEOUT (30*1000) // 30 seconds
+#define ESP_SESSION_TIMEOUT (600) // 10 Mins
+
+static const char* const SESSION_ID_COOKIE = "ESPSessionID";
+static const char* const SESSION_AUTH_COOKIE = "ESPAuthSession";
+
+//xpath in dali
+static const char* const PathSessionRoot="Sessions";
+static const char* const PathSessionProcess="Process";
+static const char* const PathSessionApplication="Application";
+static const char* const PathSessionDomain="Domain";
+static const char* const PathSessionSession="Session";
+static const char* const PropSessionID = "@id";
+static const char* const PropSessionUserID = "@userid";
+static const char* const PropSessionNetworkAddress = "@netaddr";
+static const char* const PropSessionState = "@state";
+static const char* const PropSessionCreateTime = "@createtime";
+static const char* const PropSessionLastAccessed = "@lastaccessed";
+static const char* const PropSessionLoginURL = "@loginurl";
+/* The following is an example of session data stored in Dali.
+ <Sessions>
+ <Process name="myesp">
+  <Domain name="default">
+   <Application port="8010">
+    <Session createtime="1497376914"
+             id="3831947145"
+             lastaccessed="1497377015"
+             loginurl="/"
+             netaddr="10.176.152.200"
+             state="1"
+             userid="TheAdmin"/>
+    <Session createtime="1497377427"
+             id="4106750941"
+             lastaccessed="1497377427"
+             loginurl="/"
+             netaddr="10.176.152.200"
+             state="0"/>
+   </Application>
+   <Application port="8002">
+    <Session createtime="1497376989"
+             id="3680948651"
+             lastaccessed="1497377003"
+             loginurl="/"
+             netaddr="10.176.152.200"
+             state="1"
+             userid="TheAdmin"/>
+   </Application>
+  </Domain>
+ </Process>
+</Sessions>
+ */
```

Comment:
I suspect the solution to avoid this potentially costly process, is to mangle the session id into the name, i.e. instead of
<Session id="12345">
Add:
<Session_12345>

Where you need to lookup all Sessions fo an application, you would go a getElements("Session_*")


@wangkx - can you address this/discuss this after the rest of the review is complete, then we can squash the rest and address this last issue.